oak_functions/wasm/src/lib.rs

```diff
@@ -292,19 +307,113 @@ where
     }
 }
 
+pub struct OakCaller<'a, L: OakLogger> {
+    caller: wasmi::Caller<'a, UserState<L>>,
+}
+
+pub struct Src {
+    ptr: AbiPointer,
+    len: AbiPointerOffset,
+}
+
+pub struct Dest {
+    ptr_ptr: AbiPointer,
+    len_ptr: AbiPointer,
+    buf: Vec<u8>,
+}
+
+impl<L> OakCaller<'_, L>
+where
+    L: OakLogger,
+{
+    fn read_buffer(
+        &mut self,
+        buf_ptr: AbiPointer,
+        buf_len: AbiPointerOffset,
+    ) -> Result<Vec<u8>, OakStatus> {
+        let mut memory = self.get_memory()?;
+        read_buffer(
+            &mut self.caller,
+            &mut memory,
```

Comment:
If we are already passing `caller` to this function, why do we need to find the memory here too? Couldn't `read_buffer` do that?

Reply:
`read_buffer` in 335 changes from `Caller` to `ctx : &mut impl AsContext` from which I cannot get the `memory` anymore. 

The only reason I need the second `read_buffer` is for testing, as I cannot get a `Caller` to test with (but a `Store`, which also implements `AsContext`).

I am not very happy with that and think of removing the tests (and inline the second `read_buffer`).

WDYT?